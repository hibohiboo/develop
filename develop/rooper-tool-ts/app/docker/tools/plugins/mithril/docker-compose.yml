version: '3'
services:

  # 構文チェック(ts)
  tslint:
    build: ./webpack
    volumes: 
      - ./config/tsconfig.json:/app/tsconfig.json
      - ./config/tslint.json:/app/tslint.json
      - ../../../../src/browser/plugins/ScenarioMaker/mithril:/app/src/ScenarioMaker
      - ../../../../src/browser/plugins/ScenarioMaker/common:/app/src/common
  # ビルドツール
  webpack:
    build: ./webpack
    volumes:
      - ./config/.babelrc:/app/.babelrc
      - ./config/tsconfig.json:/app/tsconfig.json
      - ./config/webpack.config.babel.js:/app/webpack.config.babel.js
      - ../../../../src/browser/plugins/ScenarioMaker/mithril:/app/src/ScenarioMaker
      - ../../../../src/browser/plugins/ScenarioMaker/common:/app/src/common
      - ../../../../dist/bundle-webpack:/app/dist
      - ../../../../src/browser/plugins/ScenarioMaker/public:/app/public
      # 構文チェック用設定
      - ./config/tslint.json:/app/tslint.json
    command: [npm, run, dev-server]
    ports:
      -  8080:8080

  # ユニットテスト用 jest
  jest:
    build: ./jest
    volumes:
      - ../../../../src/plugins/ScenarioMaker/mithril:/app/src/ScenarioMaker
      - ./jest/package.json:/app/package.json
      - ./config/tsconfig.json:/app/tsconfig.json
      - ../../../../dist/coverage-jest:/app/coverage
