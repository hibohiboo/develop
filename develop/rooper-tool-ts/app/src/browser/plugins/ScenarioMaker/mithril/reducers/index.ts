import { combineReducers } from 'redux';
import tragedySetList from './tragedySetList';

export default combineReducers({
  tragedySetList,
});
