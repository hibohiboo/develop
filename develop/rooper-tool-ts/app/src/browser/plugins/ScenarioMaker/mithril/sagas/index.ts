import { call, put, takeEvery } from 'redux-saga/effects';
import { FAILED, REQUEST, SUCCESS } from '../actions/tragedySet';
import { get as getSet } from '../browser/request';
import ITragedySetListItem from '../interfaces/ITragedySetListItem';

// ワーカー Saga: TRAGEDYSET_FETCH_REQUESTED Action によって起動する
function* fetchTragedySet(action: {type: string; payload: {url: string};}) {
  try {
    const tragedySetList: ITragedySetListItem[] = yield call(getSet, action.payload.url);
    yield put({ type: SUCCESS, payload:{ tragedySetList } });
  } catch (e) {
    yield put({ type: FAILED, message: e.message });
  }
}

/*
  TRAGEDYSET_FETCH_REQUESTED Action が送出されるたびに fetchTragedySet を起動します。
  ユーザ情報の並列取得にも対応しています。
*/
function* mySaga() {
  yield takeEvery(REQUEST, fetchTragedySet);
}

export default mySaga;
