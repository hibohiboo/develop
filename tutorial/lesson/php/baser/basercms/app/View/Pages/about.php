<!-- BaserPageTagBegin -->
<?php $this->BcBaser->setTitle('会社案内') ?>
<?php $this->BcBaser->setDescription('') ?>
<?php $this->BcBaser->setPageEditLink(2) ?>
<!-- BaserPageTagEnd -->

<h2>会社案内</h2>

<h3>会社データ</h3>

<table>
	<tbody>
		<tr>
			<th>会社名</th>
			<td>コンテストサンプル</td>
		</tr>
		<tr>
			<th>設立</th>
			<td>2015年6月</td>
		</tr>
		<tr>
			<th>所在地</th>
			<td>住所が入ります。住所が入ります。住所が入ります。</td>
		</tr>
		<tr>
			<th>事業内容</th>
			<td>事業内容１が入ります。<br />
			事業内容２が入ります。<br />
			事業内容３が入ります。</td>
		</tr>
	</tbody>
</table>

<h3>アクセスマップ</h3>
<?php $this->BcBaser->googleMaps(array("width" => 585)) ?>