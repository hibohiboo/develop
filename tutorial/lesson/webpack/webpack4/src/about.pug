extends _includes/commons/layout

block title
  title= このサイトについて

mixin link(name, url)
  a.collection-item(href=url)=name

block main
  .container
    h1.center このサイトについて
    .section
      h2 ポートレート
      p このサイトはhiboの作品の展示場です。ごゆるりと。

    .section
      h2 開発ツール
      .collection
        +link("virtual box", "https://www.virtualbox.org/") 
        +link("vagrant", "https://www.vagrantup.com/")
        +link("docker", "https://www.docker.com/")
        +link("firebase", "https://firebase.google.com/")
        +link("pug", "https://pugjs.org/api/getting-started.html")
        +link("materializecss", "https://materializecss.com/")
        +link("js開発", "https://github.com/hibohiboo/wasureta/blob/master/docker/webpack/package.json")


