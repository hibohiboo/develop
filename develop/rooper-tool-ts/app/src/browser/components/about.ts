import * as m from 'mithril';
import nav from './nav';

const component: m.Component<{}, {}> = {
  view(vnode) {
    return m('.page', [
      m(nav),
      m('h2', 'このページについて'),
      m('p', '惨劇RoopeRをオンラインで遊ぶツールです。'),
    ]);
  },
};

export default component;
