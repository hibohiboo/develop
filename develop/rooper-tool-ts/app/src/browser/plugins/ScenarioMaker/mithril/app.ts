
import * as m from 'mithril';
import { createStore } from 'redux';
import { TragedySet } from '../common/models/TragedySet';
import { requsetTragedySetList } from './actions/tragedySet';
import { get } from './browser/request';
// import InputForm from './components/InputForm';
import TragedySetForm from './components/InputForm/TragedySetForm';
import ITragedySetListItem from './interfaces/ITragedySetListItem';
import store from './store';
import InputForm from './components/InputForm';
import Provider from './mithril-redux';

class App implements m.Component<{}, {}> {
  view(vnode): Element{
    const state = store.getState() as {tragedySetList: TragedySet[]};
    const tragedySetList = state.tragedySetList;
    return m('div',[
      m(TragedySetForm, { tragedySetList }),
      m(InputForm)
    ]);
  }
 }


const root = document.getElementById('app');
function render() {
  m.render(root, m(Provider,{store}, [m(App)]));
}

// subscribeによって、dispatchのたびにrenderが呼ばれる。
store.subscribe(render);
store.dispatch(requsetTragedySetList({ url:'/tragedySets/tragedySetList.json' }));
