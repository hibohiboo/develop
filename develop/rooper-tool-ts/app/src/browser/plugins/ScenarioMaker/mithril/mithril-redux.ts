import * as m from 'mithril';

let store;

/**
 * 役割は2つ
 * 1. Mithrilコンポーネント内でreact-reduxのconnect()関数を使えるようにすること
 * 2. ラップしたコンポーネントにstore情報を渡すこと
 * 
 * @class Provider
 * @implements {m.Component<{}, {}>}
 */
export default class Provider implements m.Component<{}, {}> {
  private props;
  private app;
  
  constructor(vnode){
    store = vnode.attrs.store;
    this.props = {
      state: store.getState()
    };
    console.log('mithril-redux-const')
    console.log(vnode)
    this.app = vnode.children[0].children;
  }
  /**
   * 
   * 
   * @param {Vnode} vnode 
   * @returns 
   * @memberof Provider
   */
  view() {
    const app = this.app;
    return m(app, {
      props: {
        state: store.getState()
      }
    })
  }
}