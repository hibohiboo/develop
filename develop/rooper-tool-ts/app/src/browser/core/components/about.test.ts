
import * as m from 'mithril';
import { tidy } from 'mithril-jest';
import about from './about';

const page = {
  view: () =>
    m('div', [m(about)]),
};

describe('About component', () => {
  it('snapshot', () => {
    const cmp = m(page);
    const html = tidy(cmp);
    expect(html).toMatchSnapshot();
  });
  it('title should be h2 with label \'このページについて\'', () => {
    const cmp = m(page);
    const html = tidy(cmp);
    expect(html).toContain('このページについて');
  });
});
