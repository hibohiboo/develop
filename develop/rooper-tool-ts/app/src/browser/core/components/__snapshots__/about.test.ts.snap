// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component snapshot 1`] = `
"<div>
  <div class=\\"page\\">
    <div>
      <a href=\\"#!/\\">Home</a> <span>|</span> <a href=
      \\"#!/about\\">About</a> <span>|</span> <a href=
      \\"#!/scenariomaker\\">シナリオ作成</a>
    </div>
    <h2>
      このページについて
    </h2>
    <p>
      惨劇RoopeRをオンラインで遊ぶツールです。
    </p>
  </div>
</div>
"
`;
